src/main/java/com/xpandit/plugins/xrayjenkins/services/enviromentvariables/XrayEnvironmentVariableSetter.java

```diff
@@ -0,0 +1,183 @@
+                                                                                   final PrintStream logger) {
+        final XrayEnvironmentVariableSetter variableSetter = new XrayEnvironmentVariableSetter();
+
+        variableSetter.newVariables.put(XrayEnvironmentVariable.XRAY_RAW_RESPONSE, getRawResponses(results));
+        variableSetter.newVariables.put(XrayEnvironmentVariable.XRAY_IS_REQUEST_SUCCESSFUL, isUploadSuccessful(results));
+        variableSetter.newVariables.put(XrayEnvironmentVariable.XRAY_ISSUES_MODIFIED, getImportedFeatureIssueKeys(results, hostingType, logger));
+
+        return variableSetter;
+    }
+
+    /**
+     * Parses all the Upload Results of a Test result importation.
+     *
+     * @param results the request results.
+     * @param hostingType the hosting type of the Jira instance.
+     * @param logger the logger that will be used to log some messages.
+     * @return the XrayEnvironmentVariableSetter will all the relevant information.
+     */
+    public static XrayEnvironmentVariableSetter parseResultImportResponse(final Collection<UploadResult> results,
+                                                                          final HostingType hostingType,
+                                                                          final PrintStream logger) {
+        if (results == null) {
+            return failed();
+        }
+
+        final XrayEnvironmentVariableSetter variableSetter = new XrayEnvironmentVariableSetter();
+
+        variableSetter.newVariables.put(XrayEnvironmentVariable.XRAY_RAW_RESPONSE, getRawResponses(results));
+        variableSetter.newVariables.put(XrayEnvironmentVariable.XRAY_IS_REQUEST_SUCCESSFUL, isUploadSuccessful(results));
+
+        final String testExecKeys = getModifiedTestExecutionsKeys(results, hostingType, logger);
+        final String testKeys = getModifiedTestKeys(results, hostingType, logger);
+        variableSetter.newVariables.put(XrayEnvironmentVariable.XRAY_TEST_EXECS, testExecKeys);
+        variableSetter.newVariables.put(XrayEnvironmentVariable.XRAY_TESTS, testKeys);
+        variableSetter.newVariables.put(XrayEnvironmentVariable.XRAY_ISSUES_MODIFIED, getAllKeys(testExecKeys, testKeys));
+
+        return variableSetter;
+    }
+
+    /**
+     * Quick method that will return a Success XrayEnvironmentVariableSetter with no raw response.
+     *
+     * @return Success XrayEnvironmentVariableSetter with no raw response.
+     */
+    public static XrayEnvironmentVariableSetter success() {
+        return success(null);
+    }
+
+    /**
+     * Quick method that will return a Success XrayEnvironmentVariableSetter with a raw response.
+     *
+     * @return Success XrayEnvironmentVariableSetter with a raw response.
+     */
+    public static XrayEnvironmentVariableSetter success(@Nullable final String message) {
+        final XrayEnvironmentVariableSetter variableSetter = new XrayEnvironmentVariableSetter();
```

Comment:
Duplicated code. Can it be refactored?

Reply:
It could be refactored, but I think it will just make the code harder to understand/read. What do you think?

Reply:
If it'll make less readable, for me you can mark this conversation as resolved